### ISSUE TITLE
Allow `release_bullets()` to go in any section

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/usethis#1818]
It is currently possible to add custom bullets to the checklist generated by `usethis::use_release_issue()` through the creation of an unexported `release_bullets()` function.

But the release issue checklist contains several sections:

- First release (in some cases)
- Prepare for release
- Submit to CRAN
- Wait for CRAN...

and it's only possible to add new items to "Prepare for release".

But one may want to add custom item in other sections. In particular, in the last section, post-CRAN acceptance. One example could be "share it on slack". Or before the first release: "verify that the package is set up for upload on zenodo".

I'm not completely sure what the best way to implement this would be. My initial idea would be for `release_bullets()` to return a named list of vectors instead of a single vector but it would be a breaking change.

**Comment by jennybc** [comment]
> I'm not completely sure what the best way to implement this would be

Yeah that is the dilemma. I'm not exactly keen to enter into a formal "contract" about the sections, which would be necessary to allow custom bullets to be inserted into specific sections.

Maybe the right thing to do is to place custom bullets in their own section, knowing that a maintainer can edit the checklist and move them, if they wish. I know that's not ideal, but it might be the right compromise.

**Comment by hadley** [comment]
Maybe we could base them on the heading name, and any bullets that don't match a heading just go at the end. (So we're still covered if we change the header name.)
